refinery/ui/source/js/data-set-import/controllers/file-upload-s3.js

```diff
@@ -97,17 +97,15 @@
     };
 
     vm.uploadFiles = function () {
-      if (vm.files) {
-        vm.multifileUploadInProgress = true;
-        for (var i = 0; i < vm.files.length; i++) {
-          if (vm.isFileNew(vm.files[i])) {
-            vm.uploadFile(vm.files[i]);
-            // to enable sequential uploads
-            return;  // uploadFiles() will be called again from uploadFile()
-          }
+      vm.multifileUploadInProgress = true;
+      for (var i = 0; i < vm.files.length; i++) {
+        if (vm.isFileNew(vm.files[i])) {
+          vm.uploadFile(vm.files[i]);
+          // to enable sequential uploads
+          return;  // uploadFiles() will be called again from uploadFile()
         }
-        vm.multifileUploadInProgress = false;
       }
+      vm.multifileUploadInProgress = false;
```

Comment:
This was not actually a change in your commit, but is `vm.multifileUploadInProgress = false` accurate? My sense is that we've started all the uploadFiles that are needed, but they are still running asynchronously?

Reply:
This line is reached only after all the files have finished uploading (see `return` [inside the loop above](https://github.com/refinery-platform/refinery-platform/blob/f5901cd07b3dfaa0b95626cab41af80ca71399e0/refinery/ui/source/js/data-set-import/controllers/file-upload-s3.js#L105)).